### PR TITLE
Feat: transition of contract size limit

### PR DESCRIPTION
**Pull request by achraf17**

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -430,6 +431,7 @@ type Transition struct {
 	EpochLength           uint64   `json:"epochlength,omitempty"`           // Number of blocks that should pass before pending validator votes are reset
 	BlockPeriodSeconds    uint64   `json:"blockperiodseconds,omitempty"`    // Minimum time between two consecutive IBFT or QBFT blocks’ timestamps in seconds
 	RequestTimeoutSeconds uint64   `json:"requesttimeoutseconds,omitempty"` // Minimum request timeout for each IBFT or QBFT round in milliseconds
+	ContractSizeLimit     uint64   `json:"contractSizeLimit,omitempty"`     // Maximum smart contract code size
```
Comment (baptiste-b-pegasys) [inline params/config.go]: it's not camel case for the others

File: params/config.go [hunk 2]
```diff
@@ -580,6 +583,13 @@ func (c *ChainConfig) GetMaxCodeSize(num *big.Int) int {
 			maxCodeSize = int(c.MaxCodeSize) * 1024
 		}
 	}
+	if c.Transitions != nil && len(c.Transitions) > 0 {
```
Comment (baptiste-b-pegasys) [inline params/config.go]: the nil check is not useful, len(nil) will give 0
```suggestion
	if len(c.Transitions) > 0 {
```

File: params/config.go (hunk 1 above)
Comment (achraf17) [reply params/config.go]: updated

File: params/config.go (hunk 2 above)
Comment (achraf17) [reply params/config.go]: updated

**Review by antonydenyer — APPROVED** [review]

**Review by baptiste-b-pegasys — APPROVED** [review]

**Pull request merged**